### ISSUE TITLE
keepassxc-cli Master password with special characters returns invalid

### ISSUE THREAD

**Issue by Generator** [keepassxreboot/keepassxc#1063]
## Expected Behavior

`keepassxc-cli show db.kdbx foobar` 
Type master password and show the entry password

## Current Behavior

Using a master password with `$ ! \ "` or some other special characters  keepassxc-cli always returns with invalid password.
```
keepassxc-cli show db.kdbx foobar
Insert the database password
> 
Error while parsing the database: Wrong key or database file is corrupt.
```
## Steps to Reproduce (for bugs)

1. Create a new database
2. Type a master password with `$ ! \ "` ...
3. Use `keepassxc-cli` with some action (show, merge etc...)

## Debug Info

KeePassXC - Version 2.2.0
Revision: caa49a8ef3ee28ed478192389b21d61107b3b8e0

Libraries:
- Qt 5.9.1
- libgcrypt 1.8.1

Operating system: Arch Linux
CPU architecture: x86_64
Kernel: linux 4.12.14-1-ck-core2

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

**Comment by Generator** [comment]
Just noticed, the issue isnt with the password but the keyfile.
I've created a fresh database with same password without keyfile, and was able to merge databases without keyfile
```
gnome-keyring-query get keepass | keepassxc-cli merge -s db1.kdbx db2.kdbx                                                                                                          
Insert password to unlock db1.kdbx:                                                                                                                                                   
Successfully merged the database files.
```

But with keyfile can't unlock database
```
gnome-keyring-query get keepass | keepassxc-cli merge -s -f $PWD/db.key db1.kdbx db2.kdbx
Insert password to unlock db1.kdbx:
Error while parsing the database: Wrong key or database file is corrupt.
```
 ```
gnome-keyring-query get keepass | keepassxc-cli merge -s -k $PWD/db.key db1.kdbx db2.kdbx
Insert password to unlock db1.kdbx:
Error while parsing the database: Wrong key or database file is corrupt.
```

**Comment by Generator** [comment]
Apparently only merge isn't working with keyfile, other commands work fine with keyfile and master password with special characters

```
$ gnome-keyring-query get keepass | keepassxc-cli show -k db.key db1.kdbx Sample
Insert password to unlock db1.kdbx:
   title: Sample
username:
password: password
     URL:
   Notes:
```

**Comment by Generator** [comment]
After reading #590 clear "recent history" and unchecked "load previous databases on startup" `keepassxc-cli merge` start working.

```
$ gnome-keyring-query get keepass | keepassxc-cli merge -s -k $PWD/db.key db2.kdbx db1.kdbx
Insert password to unlock db2.kdbx: 
New entry Sample2 detected. Creating it.
Successfully merged the database files.
```

Look like to be related somehow with #590, not having "previous databases on startup" solves the problem.

**Comment by louib** [comment]
@Generator I was not able to reproduce this. You say the problem might be with the key file, but the first example you give in the issue description is not using a key file at all. I also doubt the options have anything to do, as we don't use the config options from the CLI at the moment. Can you reproduce steadily with the latest version (2.2.2)? If so please update the STRs in the description, and I'll be glad to troubleshoot the issue.

**Comment by Generator** [comment]
@louib The first was with 2.2.1 didn't have the `--key-file` option.
I also doubt it has to do with any of the options, but it started working without having changed anything in the databases.

But now the issue looks to be fixed in 2.2.2. Close the issue, it appears to be fixed.
